app/src/main/java/com/eventyay/organizer/core/roleinvite/RoleInviteFragment.java

```diff
@@ -0,0 +1,108 @@
+package com.eventyay.organizer.core.roleinvite;
+
+import android.os.Bundle;
+
+import androidx.databinding.DataBindingUtil;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.lifecycle.ViewModelProviders;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+
+import com.eventyay.organizer.R;
+import com.eventyay.organizer.common.mvp.view.BaseFragment;
+import com.eventyay.organizer.core.main.MainActivity;
+import com.eventyay.organizer.databinding.FragmentRoleInviteBinding;
+import com.eventyay.organizer.ui.ViewUtils;
+
+import javax.inject.Inject;
+
+
+import static com.eventyay.organizer.ui.ViewUtils.showView;
+import static com.eventyay.organizer.utils.ValidateUtils.validateEmail;
+
+public class RoleInviteFragment extends BaseFragment implements RoleInviteView {
+
+    @Inject
+    ViewModelProvider.Factory viewModelFactory;
+
+    private FragmentRoleInviteBinding binding;
+    private ArrayAdapter<CharSequence> rolesAdapter;
+
+    private RoleInviteViewModel roleInviteViewModel;
+
+    private long eventId;
+    private long roleId;
+
+    public static RoleInviteFragment newInstance(long eventId) {
+        RoleInviteFragment fragment = new RoleInviteFragment();
+        Bundle args = new Bundle();
+        args.putLong(MainActivity.EVENT_KEY, eventId);
```

Comment:
`EVENT_KEY` can be a public constant.

Reply:
It's already a public constant, defined in `MainActivity`.

Reply:
Oh sorry, I forgot it's Java. Can't we declare the constant outside the main activity class in java?

Reply:
We can, but since that constant is used at a lot of places so it has been declared in the MainActivity.